adapters/repos/db/nodes.go

```diff
@@ -0,0 +1,96 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/semi-technologies/weaviate/entities/models"
+)
+
+// GetNodeStatuses returns the status of all Weaviate nodes.
+func (db *DB) GetNodeStatuses(ctx context.Context) ([]*models.NodeStatus, error) {
+	var nodeStatuses []*models.NodeStatus
+	for _, nodeName := range db.schemaGetter.Nodes() {
+		status, err := db.getNodeStatus(ctx, nodeName)
+		if err != nil {
+			return nil, fmt.Errorf("node: %v: %w", nodeName, err)
+		}
+		nodeStatuses = append(nodeStatuses, status)
+	}
+
+	sort.Slice(nodeStatuses, func(i, j int) bool {
+		return nodeStatuses[i].Name < nodeStatuses[j].Name
+	})
+	return nodeStatuses, nil
+}
+
+func (db *DB) getNodeStatus(ctx context.Context, nodeName string) (*models.NodeStatus, error) {
+	if db.schemaGetter.NodeName() == nodeName {
+		return db.localNodeStatus(), nil
+	}
+	status, err := db.remoteNode.GetNodeStatus(ctx, nodeName)
+	if err != nil {
+		if strings.HasPrefix(err.Error(), "open http request") || strings.HasPrefix(err.Error(), "send http request") {
```

Comment:
This could break in the future if the error string is ever changed in `adapters/clients/cluster_node.go`. Is it possible to write a test for this to at least notify of breakage?